sparrowpy/dsp.py

```diff
@@ -229,6 +228,190 @@ def dirac_sequence(
 
     return dirac_sequence
 
+def weight_filters_by_etc(
+    etc: pf.TimeData,
+    signal: pf.Signal,
+    bandwidth=None,
+) -> pf.Signal:
+    r"""
+    Generate the frequency-wise impulse response for a given ETC.
+
+    The frequency-band-wise filters are weighed by the respective
+    ETC over time after Chapter 5.3.4 of [#]_. The weighting is performed
+    after the following equation.
+
+    .. math::
+        h_i = \nu_i \cdot \sqrt{\frac{E_n(k)}{\sum^{g(k)}_{g(k-1)+1} \nu_i^2}}
+    ... \cdot \sqrt{\frac{BW}{f_s/2}}
+
+    where :math:`h_i` and :math:`\nu_i` represent respectively the
+    weighted output signal and the input signal's at a given time sample
+    :math:`i`.
+    :math:`g(k)=floor(k \cdot f_s \cdot \Delta t)` represents the range of each
+    energy window with given length :math:`\Delta t` of the ETC entry
+    :math:`E(k)` with index :math:`k`. :math:`BW` is the bandwidth
+    of the energy time curve, which is set to half the sampling rate
+    by default.
+
+    Parameters
+    ----------
+    etc: :py:class:`pyfar.TimeData'
+        ETC of a sound propagation simulation of cshape
+        ``(..., n_freq_bands)``.
+        *Note:* The ETC entries must be equally spaced in time.
+    signal: :py:class:`pyfar.Signal'
+        signal to be weighted by the etc of cshape
+        ``(..., n_freq_bands)``.
+    bandwidth: np.ndarray
+        Bandwidth array corresponding to the noise filter channels in Hz.
+        If None, signal will be processed as full spectrum
+        ``sampling_rate/2``.
+
+    Returns
+    -------
+    weighted_signal : :py:class:`pyfar.Signal'
+        signal weighted by the etc. The cshape matches the cshape of the etc.
+
+    References
+    ----------
+    .. [#] D. Schröder, “Physically based real-time auralization of
+           interactive virtual environments,” PhD Thesis, Logos-Verlag,
+           Berlin, 2011. [Online].
+           Available: https://publications.rwth-aachen.de/record/50580
+
+    .. note::
+        The weighted signal output results from a broadcast between the
+        input signal, the input ETC, and the bandwidth.
+        Thus, it's important to ensure that the right-most channel dimensions
+        of etc and signal are compatible with eachother and the bandwidths.
+        That is, ensuring etc.cshape=(...,bandwidth.shape) and
+        signal.cshape=(...,bandwidth.shape)
+
+    .. note::
+        The input of this function follows the sparrowpy convention for etcs,
+        where the dimension which represents the different frequency bands is
+        the right-most dimension of etc.cshape=(..., n_frequency_bands).
+        The input signal must also reflect this convention.
+        This is the opposite from the pyfar convention for fractional octave
+        band filtered signals, where
+        filtered_signal.cshape=(n_frequency_bands,...).
+
```

Comment:
```suggestion
```

i think we dont need to clarify this, we just do it, its ducumented in detial, what is needed and what not.